### PR TITLE
try use compression added to http client options

### PR DESCRIPTION
**Pull request by wsatyla**
This is fix for https://github.com/Cognifide/knotx/issues/33

I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

File: knotx-core/src/main/java/com/cognifide/knotx/repository/RepositoryType.java [hunk 1]
```diff
@@ -44,7 +46,7 @@ public boolean validate(RepositoryConfiguration.RepositoryMetadata metadata) {
         public Repository<String, URI> create(RepositoryConfiguration.RepositoryMetadata metadata,
                                               KnotxVerticle verticle) {
             return RemoteRepository.of(metadata.getPath(), metadata.getDomain(), metadata.getPort(), verticle
-                    .getVertx().createHttpClient());
+                    .getVertx().createHttpClient(new HttpClientOptions().setTryUseCompression(true)));
```
Comment (Skejven) [inline knotx-core/src/main/java/com/cognifide/knotx/repository/RepositoryType.java]: This option should be configurable.

**Comment by Skejven** [comment]
Please add proper pull request message according to [Contributing Rules](https://github.com/Cognifide/knotx/blob/master/CONTRIBUTING.md#pull-requests).

File: knotx-core/src/main/java/com/cognifide/knotx/repository/RepositoryType.java (hunk 1 above)
Comment (marcinczeczko) [inline knotx-core/src/main/java/com/cognifide/knotx/repository/RepositoryType.java]: I dont't think we need to have it configurable. As the method says it tries to use compression. if it's not supported it still work as expected.

**Comment by wsatyla** [comment]
I hereby agree to the terms of the Knot.x Contributor License Agreement.

File: knotx-core/knotx-repository/src/main/java/com/cognifide/knotx/repository/RepositoryConfiguration.java [hunk 2]
```diff
@@ -149,6 +154,14 @@ public void setType(RepositoryType type) {
             this.type = type;
         }
 
+        public JsonObject getClientOptions() {
+            return clientOptions;
+        }
+
+        public void setClientOptions(JsonObject clientOptions) {
```
Comment (marcinczeczko) [inline knotx-core/knotx-repository/src/main/java/com/cognifide/knotx/repository/RepositoryConfiguration.java]: I don't see clientOptions are being set anywere except unit test. Looks like you missed passing it from verticle.

Comment (wsatyla) [inline knotx-core/knotx-repository/src/main/java/com/cognifide/knotx/repository/RepositoryConfiguration.java]: Client options are being set at 44 line of RepositoryConfiguration:
repositoryMetadata.clientOptions = item.getJsonObject("client.options", new JsonObject());
All the setters of RepositoryConfiguration.RepositoryMetadata are used in UnitTests only

**Comment by pawelkoper** [comment]
you forgot to add 'client.options': to 'knotx-example-monolith.json'

**Comment by Skejven** [comment]
Approved.

**Pull request merged**